### ISSUE TITLE
backslash line continuation pre-object breaks objects

### ISSUE THREAD

**Issue by michael1006** [jashkenas/coffeescript#2457]
I tend to use backslash line continuation liberally, since I've been bitten by a lot of times when CoffeeScript breaks without it, for example this breaks if _any_ of the backslashes are missing:

```
switch condition
  when \
      "long first condition on one line when I want to limit line length to 80 chars", \
      "second condition on a third line", \
      "third condition, different lines for readability"
    doSomething()
```

But now I'm being bit by the opposite, I tried writing:

```
                callFunction arg1, \
                  object_key_1: data1 
                  object_key_2: data2
```

But it compiles, bizarrely, to:

```
callFunction(arg1, {
  object_key_1: data1
});

({
  object_key_2: data2
});
```

I can't imagine why it's splitting up the object. Removing the backslash produces the desired behavior:

```
                callFunction arg1, 
                  object_key_1: data1 
                  object_key_2: data2
```

produces

```
callFunction(arg1, {
  object_key_1: data1,
  object_key_2: data2
});
```

Now, backslashes aren't even a documented functionality of CoffeeScript, at least not that I can find on the website, despite the fact that they're sometimes _required_ for multi-line functionality (as in the `switch` above). And I can't figure out for the life of me _when_ they're required, and when they're not, except through trial-and-error.

But the fact that they break existing code when they're added in places where they should just be redundant, seems like a bug, unless there's something additional that backslashes mean that I'm missing.

Basically, shouldn't I be able to add a backslash to the end of _any_ line (except perhaps object property lists, and of course multiline strings), without it changing the meaning of my code? So I'm filing this as a bug... Please correct any misunderstandings I may have here, though!
